### PR TITLE
Run UAT tests on every commit on master

### PR DESCRIPTION
**Pull request by lucafavatella**
Depends on #799 (included in this PR).
The idea of this PR is running on CI more frequently UAT tests that we have so not to break what currently passes.
Sample CI workflow can be seen at https://circleci.com/workflow-run/0b1fe102-b57f-4e73-9c5b-0b59ed00ac66

In scope for https://www.pivotaltracker.com/story/show/155510659

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -333,6 +336,13 @@ workflows:
             tags:
               only: /^v.*$/
 
+      - uat_tests:
+          requires:
+            - static_analysis
```
Comment (dincho) [inline .circleci/config.yml]: What's the reason it requires only static_analysis ? Why having requirements at all?

**Comment by lucafavatella** [comment]
> Is it reasonable to deploy to integration if UAT tests fail?

Good idea. Let me integrate this.

File: .circleci/config.yml (hunk 1 above)
Comment (lucafavatella) [reply .circleci/config.yml]: Good point, let me remove this.

My initial thought was ensuring swagger generated files were checked before UAT tests. Though no need for it.

**Comment by lucafavatella** [comment]
@dincho I addressed both comments.

**Comment by lucafavatella** [comment]
As discussed I prefer spending more effort on fixing more UAT tests (e.g. https://github.com/aeternity/epoch/pull/799 ) before merging an improved version of this PR, not ignoring failing UAT tests.

**Comment by lucafavatella** [comment]
I force-pushed:
* Rebased on current master - meant to contain some test fixes by @velzevur ;
* I made UAT tests job to depend on build job;
* I temporarily enabled UAT tests on all branches - not only master - so to see current outcome of UAT tests on this PR.

**Comment by lucafavatella** [comment]
Current state of UAT tests:
```
...
Test node discovery ... 
Alice has address http://localhost:3013/ and no peers
Bob has address http://localhost:3023/ and peers [http://localhost:3013/]
Carol has address http://localhost:3033/ and peers [http://localhost:3023/]

Node dev1 starting

Node dev2 starting

Node dev3 starting
Carol now has peers ['http://localhost:3013/', 'http://localhost:3023/']
FAIL

======================================================================
FAIL: Test node discovery
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/builder/epoch/py/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/builder/epoch/py/tests/integration/test_use_cases.py", line 186, in test_node_discovery
    assert_equals(carol_top.height >= blocks_to_mine, True)
AssertionError: False != True

----------------------------------------------------------------------
XML: nosetests.xml
----------------------------------------------------------------------
Ran 13 tests in 281.730s

FAILED (failures=1)
```

**Comment by lucafavatella** [comment]
Apparently intermittent test failure in test job tracked at https://www.pivotaltracker.com/story/show/155582602/comments/186628457

**Comment by lucafavatella** [comment]
I am closing this PR. Will open new rebased improved one.

**Pull request closed without merging**